Helper/RuleConverter.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future.
+ *
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteTargetRule
+ * @author    Richard BAYET <richard.bayet@smile.fr>
+ * @copyright 2017 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+
+namespace Smile\ElasticsuiteTargetRule\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Smile\ElasticsuiteCatalogRule\Model\RuleFactory;
```

Comment:
@romainruaud Here, I use the elasticsuite-catalog-rule RuleFactory, which is replaced by DI in elasticsuite-virtualcategory by \Smile\ElasticsuiteVirtualCategory\Model\RuleFactory.
So I end up in the end with a  \Smile\ElasticsuiteVirtualCategory\Model\Rule by a fortunate accident, which means the generated search query correctly handled the virtual category at the \Smile\ElasticsuiteVirtualCategory\Model\Rule\Condition\Product::getSearchQuery level.

=> The case could be made that I should also use the elasticsuite-catalog-rule Combine/Product models in the conversion XML mappings for the conditions... 
But considering that my custom **actions** Combine/Product directly extends those of elasticsuite-virtualcategory, shouldn't I change here the dependency to make clear that the module is expected to use the Rule of that module ?